src/battle/action_cmd/hammer.c

```diff
@@ -2,12 +2,142 @@
+
+    if (battleStatus->unk_83 == 0) {
+        battleStatus->actionSuccess = 0;
+        battleStatus->unk_84 = 0;
+        return ApiStatus_DONE2;
+    }
+
+    func_80268858();
+    actionCommandStatus->actionCommandID = 2;
+    actionCommandStatus->hudElementX = -0x30;
+    actionCommandStatus->unk_4C = 0;
+    actionCommandStatus->unk_60 = 0;
+    actionCommandStatus->hudElementY = 0x60;
+
+    hud_element = create_hud_element(&D_802922F0);
+    actionCommandStatus->hudElements[0] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0xA);
+
+    hud_element = create_hud_element(&D_8029275C);
+    actionCommandStatus->hudElements[1] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0);
+
+    hud_element = create_hud_element(&D_802927F0);
+    actionCommandStatus->hudElements[2] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0);
+
+    hud_element = create_hud_element(&D_80292848);
+    actionCommandStatus->hudElements[3] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0);
+
+    hud_element = create_hud_element(&D_802928A0);
+    actionCommandStatus->hudElements[4] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0);
+
+    hud_element = create_hud_element(&D_80108E48);
+    actionCommandStatus->hudElements[5] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0);
+
+    hud_element = create_hud_element(&D_80292A2C);
+    actionCommandStatus->hudElements[6] = hud_element;
+    set_hud_element_flags(hud_element, 0x82);
+    set_hud_element_render_pos(hud_element, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hud_element, 0);
+    set_hud_element_alpha(hud_element, 0xFF);
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A9258_422258(ScriptInstance* script, s32 isInitialCall) {
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    BattleStatus* battleStatus = &gBattleStatus;
+    s32 *ptrReadPos = script->ptrReadPos;
```

Comment:
We call this `args` and we give it the type `Bytecode*` instead of s32*. Though I think we talked about removing bytecode @nanaian ? should we do that still or how do you feel about it? I'm kinda pro